staging/src/k8s.io/component-base/logs/json/json_test.go

```diff
@@ -109,12 +123,22 @@ func TestZapLoggerV(t *testing.T) {
 		return time.Date(1970, time.January, 1, 0, 0, 0, 123, time.UTC)
 	}
 
-	for i := 0; i < 11; i++ {
+	maxV := 12
+	for i := 0; i < maxV; i++ {
 		var buffer bytes.Buffer
 		writer := zapcore.AddSync(&buffer)
-		sampleInfoLogger, _ := NewJSONLogger(writer, nil, nil)
+		sampleInfoLogger, _ := NewJSONLogger(config.VerbosityLevel(maxV-2), writer, nil, nil)
 		sampleInfoLogger.V(i).Info("test", "ns", "default", "podnum", 2, "time", time.Microsecond)
 		logStr := buffer.String()
+
+		if i+1 == maxV {
```

Comment:
I guess you want to start testing passing verbosity to logger instance, however they way it's implemented it's not obvious. Instead of using `maxV-2`, can you create variable `expectedVerbosity` and also change the condition to:
```
if logged := logStr != ""; shouldLogged := i < expectedVerbosity;  logged != shouldLogged {
  t.Fatal(...)
```

Reply:
I've updated the test along those lines. I also renamed other variables to make it clearer.